bite-size-notes/week-2-d-Intro-To-Convex-Optimization.md

```diff
@@ -0,0 +1,143 @@
+
+## Scope and Objective
+These notes will serve as a brief introduction to Convex Optimization within the field of Controls. Since this is also my introduction to optimization, it will cover more of the basics of convex optimization and what distinguishes it from other optimization problems.  
+
+## Introduction to optimization
+Optimization is a topic that is used in mathematics, computer science, engineering, and economics. It is a method of finding the best or optimal solution to a defined problem or system. Optimization problems can occur on an incredibly broad scale, going from optimizing everyday things like coffee consumption based on your relaxed constraints in your everyday life to optimizing aircraft performance based on 500 strict mechanical, aerodynamic, and financial constraints. Almost any system can be made into an optimization problem, and due to advancements in mathematical computing, most problems can be solved using a standard method and code base.
+
+In controls, optimization allows us not just to choose a feasible control input for our system, or even just a good control for our system, but to actually find the best control input for our system based on our constraints, desired behavior, and current state. This can be a powerful tool. However, there are some limiting factors. While the mathematical framework for setting up an optimization problem can be used to define most problems, actually solving a problem can yield very crude solutions. The key to solving an optimization problem simply and convincingly is to frame the problem using a convex objective function and convex subset[3]. The rest of these notes will explain what convex functions, subsets, and optimizations are and how to recognize and build systems that fit within them.   
+
+
+## Preliminaries  
+
+### Convexity 
+Mathematics defines a function as convex if a line segment can be drawn between any two distinct points, and the rest of the function stays below that line.  
+
+<p align="center">
+  <img src="figs/Convex_vs._Not-convex.jpg" alt="drawing" width="500"/>
+</p> 
+[4]
+
+The function is usually visually shaped like a bowl or a U. 
+
+The significance of convexity in optimization is primarily due to the property of convex functions to have a single global minimum. Since a Convex function is bowl-shaped, everything is sloped to a single lowest point. This means there is only one optimal solution, and all paths lead towards it. You can imagine that for a function with many dips and valleys, it may be challenging to find the global minimum since there are so many local minimums. 
+
+**Convex Set:** Contains a line segment between any two points in the set.  
+
+ex:   
+<p align="center">
+  <img src="figs/ConvexExamples.jpg" alt="drawing" width="500"/>
+</p>  
+[1]
+The first shape is a convex set, while the other two are not convex.  
```

Comment:
1. Add an explanation for why the third shape is not convex. The reason is because it does not include its boundary.